src/TumblThree/TumblThree.Applications/Crawler/TumblrLikedByCrawler.cs

```diff
@@ -150,7 +150,10 @@ private async Task CrawlPageAsync(int crawlerNumber)
                     string url;
                     try
                     {
-                        url = nextPage.Take(Ct);
+                        if (!nextPage.TryTake(out url))
```

Comment:
Did an error pop up here or do you just not like the Take method?
By using this overload of the TryTake method, could a side effect occur?

Reply:
>  ... but in this case there're no other threads...

What about the tasks in the calling method? Could we use another overload of the `TryTake` method?
However, upon closer inspection, one might suspect that there may be another problem preventing it from executing as originally intended.